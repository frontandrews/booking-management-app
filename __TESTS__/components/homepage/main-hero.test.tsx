import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import MainHero from '@/components/homepage/main-hero';

jest.mock('next/link', () => {
  const Link = ({ children, ...rest }: any) => <a {...rest}>{children}</a>;

  return Link;
});

describe('MainHero Component', () => {
  test('renders the main hero component correctly', () => {
    const { container } = render(<MainHero />);

    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent(
      'Vacation Rental Software',
    );

    expect(screen.getByText(/Do less & earn more!/i)).toBeInTheDocument();

    const link = screen.getByRole('link', { name: /get started/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/sign-in');

    expect(container).toMatchSnapshot();
  });

  test('applies correct classes to elements', () => {
    const { container } = render(<MainHero />);

    expect(screen.getByRole('heading', { level: 1 })).toHaveClass(
      'text-4xl font-bold tracking-tight text-white sm:text-6xl',
    );

    expect(screen.getByText(/Do less & earn more!/i)).toHaveClass(
      'mt-6 text-lg leading-8 text-gray-300',
    );

    expect(container).toMatchSnapshot();
  });
});
