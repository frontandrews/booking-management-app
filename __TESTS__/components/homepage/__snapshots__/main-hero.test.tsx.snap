// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHero Component applies correct classes to elements 1`] = `
<div>
  <div
    class="bg-primary"
  >
    <div
      class="relative isolate overflow-hidden pt-14"
    >
      <div
        aria-hidden="true"
        class="absolute inset-x-0 -top-40 -z-10 transform-gpu overflow-hidden blur-3xl sm:-top-80"
      />
      <div
        class="slide-up mx-auto max-w-4xl py-32 sm:py-48 lg:py-56"
      >
        <div
          class="text-center"
        >
          <h1
            class="text-4xl font-bold tracking-tight text-white sm:text-6xl"
          >
            Vacation Rental Software
          </h1>
          <p
            class="mt-6 text-lg leading-8 text-gray-300"
          >
            Do less & earn more! Our app helps property managers scale and grow their businesses with vacation the best rental management software.
          </p>
          <div
            class="mt-10 flex items-center justify-center gap-x-6"
          >
            <a
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
              href="/sign-in"
            >
              Get started
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainHero Component renders the main hero component correctly 1`] = `
<div>
  <div
    class="bg-primary"
  >
    <div
      class="relative isolate overflow-hidden pt-14"
    >
      <div
        aria-hidden="true"
        class="absolute inset-x-0 -top-40 -z-10 transform-gpu overflow-hidden blur-3xl sm:-top-80"
      />
      <div
        class="slide-up mx-auto max-w-4xl py-32 sm:py-48 lg:py-56"
      >
        <div
          class="text-center"
        >
          <h1
            class="text-4xl font-bold tracking-tight text-white sm:text-6xl"
          >
            Vacation Rental Software
          </h1>
          <p
            class="mt-6 text-lg leading-8 text-gray-300"
          >
            Do less & earn more! Our app helps property managers scale and grow their businesses with vacation the best rental management software.
          </p>
          <div
            class="mt-10 flex items-center justify-center gap-x-6"
          >
            <a
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
              href="/sign-in"
            >
              Get started
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
